src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/Components/ComponentGenericTypePass.cs

```diff
@@ -460,17 +460,13 @@ private void CreateTypeInferenceMethod(DocumentIntermediateNode documentNode, Co
 
             var classNode = namespaceNode.Children
                 .OfType<ClassDeclarationIntermediateNode>()
-                .FirstOrDefault(n => n.ClassName == "TypeInference");
+                .FirstOrDefault(n => n.Name == "TypeInference");
             if (classNode == null)
             {
                 classNode = new ClassDeclarationIntermediateNode()
                 {
-                    ClassName = "TypeInference",
-                    Modifiers =
-                        {
-                            "internal",
-                            "static",
-                        },
+                    Name = "TypeInference",
+                    Modifiers = ["internal", "static"]
```

Comment:
>["internal", "static"] [](http://example.com/codeflow?start=32&length=22)

Probably not a significant allocation, but this and other places where we set the modifiers to a known set, could use a static instead of allocating an array each time.